hipcheck/src/policy_exprs/env.rs

```diff
@@ -347,7 +347,11 @@ fn gt(env: &Env, args: &[Expr]) -> Result<Expr> {
 				.map_err(|err| Error::Datetime(err.to_string()))?
 				.is_gt(),
 		)),
-		_ => unreachable!(),
+		(a, b) => Err(Error::MismatchedTypes(
+			Box::new(a),
+			Box::new(b),
+			name.to_string(),
+		)),
```

Comment:
I think I would suggest reverting all of these changes associated with the `MismatchedTypes` variant, and remove the variant. The PR #514 will make this change obsolete, and merging it now will just cause conflicts with that PR.